test_runner/batch_others/test_broken_timeline.py

```diff
@@ -72,9 +72,11 @@ def test_broken_timeline(neon_env_builder: NeonEnvBuilder):
 
     # But all others are broken
 
-    # First timeline would fail instantly due to corrupt metadata file
+    # First timeline would not get loaded into pageserver due to corrupt metadata file
```

Comment:
One of the biggest implications is this, sort of related to https://github.com/neondatabase/neon/pull/2352

Now we can have a situation where a timeline could exist locally, but never get to be loaded into memory, even after a restart, due to corrupt metadata in it. That might influence the fact that we'll never be able to init such tenant again, currently returning 500.